javav2/example_code/transcribe-streaming/src/main/java/com/amazonaws/transcribestreaming/Microphone.java

```diff
@@ -1,27 +1,27 @@
-// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-// SPDX-License-Identifier: Apache-2.0
-
-package com.amazonaws.transcribe;
-
-// snippet-start:[transcribe.java2.bidir_streaming_microphone.complete]
-
-// snippet-start:[transcribe.java2.bidir_streaming_microphone.import]
-import javax.sound.sampled.AudioFormat;
-import javax.sound.sampled.AudioSystem;
-import javax.sound.sampled.DataLine;
-import javax.sound.sampled.TargetDataLine;
-// snippet-end:[transcribe.java2.bidir_streaming_microphone.import]
-
-public class Microphone {
-
-    public static TargetDataLine get() throws Exception {
-        AudioFormat format = new AudioFormat(16000, 16, 1, true, false);
-        DataLine.Info datalineInfo = new DataLine.Info(TargetDataLine.class, format);
-
-        TargetDataLine dataLine = (TargetDataLine) AudioSystem.getLine(datalineInfo);
-        dataLine.open(format);
-
-        return dataLine;
-    }
-}
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.transcribestreaming;
+
+// snippet-start:[transcribe.java2.bidir_streaming_microphone.complete]
+
+// snippet-start:[transcribe.java2.bidir_streaming_microphone.import]
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.DataLine;
+import javax.sound.sampled.TargetDataLine;
+// snippet-end:[transcribe.java2.bidir_streaming_microphone.import]
+
+public class Microphone {
+
+    public static TargetDataLine get() throws Exception {
+        AudioFormat format = new AudioFormat(16000, 16, 1, true, false);
+        DataLine.Info datalineInfo = new DataLine.Info(TargetDataLine.class, format);
+
+        TargetDataLine dataLine = (TargetDataLine) AudioSystem.getLine(datalineInfo);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *dataLine*. The resource is referenced at lines 21, 22. The resource is returned at line 24. There are other execution paths that don&#39;t close the resource or return it, for example, when TargetDataLine.open throws an exception. To prevent this resource leak, close *dataLine* along these other paths before you exit this method.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).